### PR TITLE
Unit.swift Store units

### PR DESCRIPTION
**Pull request by e78l**
Reduced code size some w/single initializer in Dimension extension (vs 1 init/Unit* class). Added to size with big enum (stats off Ubuntu below) and other minor changes.

This would be a first step in the Measurement+DateComponents formatter implementation I would like to propose: e78l@613b64bc5e4f9f185c52ba260e8610930c6882b1

After PR:
   text	   data	    bss	    dec	    hex	filename
8773109	 386873	 151560	9311542	 8e1536	libFoundation.so

Before PR, off 17912deecfd3b096857cfa08bdcf6cb2c7c3eb30
   text	   data	    bss	    dec	    hex	filename
8711549	 386873	 151528	9249950	 8d249e	libFoundation.so

### CODE REVIEW

File: Foundation/Unit.swift [hunk 1]
```diff
@@ -147,11 +148,226 @@ private class UnitConverterReciprocal : UnitConverter, NSSecureCoding {
...
+        case quart
+        case tablespoon
+        case teaspoon
+
+        static let max = teaspoon.rawValue + 1
+    }
+
+    // Store is private, access members by `stored(at:)`
+    // Set members by `stored(_,at:)`
+    private static var ObjectStore = [Unit?](repeating: nil, count: StoreIndex.max)
```
Comment (millenomi) [inline Foundation/Unit.swift]: Style: lowercase. `objectStore`. Maybe `cachedDimensions`, `storedDimensions`?

File: Foundation/Unit.swift [hunk 2]
```diff
@@ -147,11 +148,226 @@ private class UnitConverterReciprocal : UnitConverter, NSSecureCoding {
...
+
+    fileprivate static func stored(at index: StoreIndex) -> Unit? {
+        return ObjectStore[index.rawValue]
```
Comment (millenomi) [inline Foundation/Unit.swift]: These aren't thread-safe, and likely need a lock.

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@e78l Are you still interested in this patch?

**Comment by millenomi** [comment]
@swift-ci please test Linux

**Comment by millenomi** [comment]
@swift-ci please test Linux

**Comment by e78l** [comment]
Hoping to push the changes this week

**Comment by shahmishal** [comment]
The Swift project moved the default branch to `main` and deleted `master` branch, so GitHub automatically closed the PR. Please re-create pull request with main branch.

More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Pull request closed without merging**